### PR TITLE
Added Download Service

### PR DESCRIPTION
**Pull request by aggarwalpulkit596**

### CODE REVIEW

File: app/src/main/java/com/codingblocks/cbonlineapp/services/DownloadService.kt [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    override fun onHandleIntent(intent: Intent?) {
+
+        notificationManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
+        notificationBuilder = Notification.Builder(this)
+                //                .setSmallIcon(R.drawable.ic_download)
+                .setContentTitle("Download")
+                .setContentText("Downloading File")
+                .setAutoCancel(true)
+        notificationManager!!.notify(0, notificationBuilder!!.build())
```
Comment (the-dagger) [inline app/src/main/java/com/codingblocks/cbonlineapp/services/DownloadService.kt]: use ?.let{} here, that's a better practice than using !!

File: app/src/main/java/com/codingblocks/cbonlineapp/services/DownloadService.kt [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        output.flush()
+        output.close()
+        bis.close()
+
+    }
+
+    private fun sendNotification(download: Download) {
+
+        sendIntent(download)
+        notificationBuilder!!.setProgress(100, download.progress, false)
```
Comment (the-dagger) [inline app/src/main/java/com/codingblocks/cbonlineapp/services/DownloadService.kt]: Here as well

File: onlineapi/src/main/kotlin/com/codingblocks/onlineapi/models/APIModel.kt [hunk 3]
```diff
@@ -529,6 +530,13 @@ class Announcement : BaseModel() {
     var run_id: String? = null
 
 }
+class Download : Serializable {
```
Comment (the-dagger) [inline onlineapi/src/main/kotlin/com/codingblocks/onlineapi/models/APIModel.kt]: Maybe we can use Parcelable instead of Serializable?

**Comment by aggarwalpulkit596** [comment]
@championswimmer i've added the json as well file is that okay ?

**Pull request merged**